app/controllers/medications_controller.rb

```diff
@@ -0,0 +1,75 @@
+require 'nokogiri'
+require 'open-uri'
+
+class MedicationsController < ApplicationController
+  def new
+  end
+
+  def search
+    response = MSET_SERVICE.med_search(params[:medication_name])
+    json = json_parse(response)
+
+    @med_hash = Hash.new(0)
+    if json[:results].nil?
+      redirect_to '/medications/new'
+      flash[:warning] = "Sorry, your search did not return any results. Please try another search."
+    else
+      json[:results].each do |result|
+        if @med_hash.keys.include?(result[:brand_name])
+          next
+        else
+          @med_hash[result[:brand_name]] = result[:product_ndc]
+        end
+      end
+    end
+  end
+
+  def create
+    medication = current_user.medications.create(brand_name: med_params[:name], generic_name: 'unknown', product_ndc: med_params[:product_ndc])
+
+    # GETTING SYMPTOMS
+
+    response = MSET_SERVICE.sym_search(medication.product_ndc)
+    json = json_parse(response)
+
+    tables = json[:results].map do |result|
+      result[:adverse_reactions_table]
+    end
+
+    unless tables[0].nil?
+      symptoms = []
+      tables.each do |table|
+        table.each do |t|
+          page = Nokogiri::XML(t)
+          page.css('tbody').select do |node|
+            node.traverse do |el|
+              # require "pry"; binding.pry if el.name == 'footnote'
+              symptoms << el.text.strip unless el.text.include?('%') || el.text.include?('System') || el.text.strip == 'General' || el.text.strip == 'Metabolic/Nutritional' || el.name == 'footnote' || el.text == ' ' || el.text.split(' ').size > 3 || el.text.include?('only') || el.text=~ /\d/
```

Comment:
This needs to be thoroughly refactored, and there are already additional conditions that I want to add to this.  Just FYI to everyone.

Reply:
@ejdelsztejn refactored all this code; the Nokogiri section now lives in a helper method under `SearchResultsFacade#nokogiri_parser` in case you want to change it up later. That method gets called inside `#extract_symptoms` in the same class.